src/gyro/Gyro.ts

```diff
@@ -0,0 +1,136 @@
+
+
+/**
+ * Introducing Gyro (gyroerometer, on mobile) support for WebChucK. Gyro wraps
+ * JavaScript DeviceMotionEvent listeners easing access to mobile device gyroerometers
+ * in WebChucK code. 
+ *
+ * To get started with Gyro:
+ * @example
+ * ```ts
+ * import { Chuck, Gyro } from "webchuck";
+ *
+ * const theChuck = await Chuck.init([]);
+ * const gyro = await Gyro.init(theChuck); // Initialize Gyro
+ * ```
+ * 
+ * The "deviceorientation" event gives motion of the device around the three axes (x, y, and z) represented as degrees from 0 to 360.
+ * More on the "deviceorientation" event can be found online {@link https://developer.mozilla.org/en-US/docs/Web/API/Window/deviceorientation_event | here }.
+ * 
+ * iOS devices require that the web developer ask permission from the user to access sensors after a button push. This looks like: 
+ * 
+ * ```ts
+ * import { Chuck, Gyro} from '../webchuck/src/wc-bundle.js';
+ * 
+ * let theChuck = await Chuck.init([]); // context suspended
+ * let gyro = await Gyro.init(theChuck);
+ * 
+ * let runButton = document.getElementById("run");
+ * runButton.disabled = false;
+ * 
+ * runButton.addEventListener("click", async () => {
+ *   if (typeof DeviceOrientationEvent.requestPermission === 'function') {
+ *     await DeviceOrientationEvent.requestPermission();
+ *   } 
+ * 
+ *   await theChuck.loadFile("./yourChuckCode.ck");
+ *   theChuck.runFile("yourChuckCode.ck");
+ * 
+ * });
+ * ```
+ */
+export default class Gyro {
+  // Private members
+  private theChuck: Chuck;
+  private _gyroActive: boolean = false;
+
+  private boundHandleOrientation;
+
+
+  /** @internal */
+  constructor(theChuck: Chuck) {
+    // Initialize members
+    this.theChuck = theChuck;
+
+    this.boundHandleOrientation = this.handleOrientation.bind(this);
+  }
+
+  /**
+   * Initialize Gyro functionality in your WebChucK instance.
+   * This adds a `Gyro` and `GyroMsg` class to the ChucK Virtual Machine (VM).
+   * Gyroerometer event (DeviceMotionEvent) listeners are added if `enableGyro` is true (default).
+   * @example
+   * ```ts
+   * theChuck = await Chuck.init([]);
+   * gyro = await Gyro.init(theChuck); // Initialize Gyro 
+   */
+  static async init(
+    theChuck: Chuck,
+    enableGyro: boolean = true
+  ): Promise<Gyro> {
+    const gyro = new Gyro(theChuck);
+    // Add Gyro and GyroMsg classes to ChucK VM
+    await gyro.theChuck.runCode(GyroMsg_ck);
+    await gyro.theChuck.runCode(Gyro_ck);
+
+    // Enable mouse and keyboard
+    gyro.enableGyro();
+    return gyro;
+  }
+
+
+
+  /**
+   * @internal
+   * Check if gyro is active
+   */
+  async gyroActive() {
+    const x = await this.theChuck.getInt("_gyroActive");
+    this._gyroActive = x == 1;
+  }
+
+  /**
+   * Enable Javascript event (DeviceMotionEvent) listeners for Gyro
+   * @example
+   * ```ts
+   * // If gyro is not yet enabled
+   * gyro.enableGyro();
+   * ```
+   */
+  enableGyro() {
+    // consider using "deviceorientationabsolute" 
```

Comment:
Did we want to still consider this functionality down the line?

Reply:
I don't think so. It isn't supported on some browsers, and seems deviceorientation is the more supported variety.